src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -573,15 +573,16 @@ private void reconfigureKafkaConsumer(final boolean forceSeek) {
                         .map(pk -> {
                             final NakadiCursor beforeFirstAvailable = beforeFirst.get(pk);
 
-                            // Checks that current cursor is still available in storage
+                            // Checks that current cursor is still available in storage. Otherwise reset to oldest
+                            // available offset for the partition
                             offsets.get(pk).ensureDataAvailable(beforeFirstAvailable);
                             return offsets.get(pk).getSentOffset();
                         })
                         .collect(Collectors.toList());
-
                 eventConsumer.reassign(cursors);
             } catch (InvalidCursorException ex) {
-                throw new NakadiRuntimeException(ex);
+                getLog().error("Failed to reassign updated cursors for subscription", ex);
+                switchState(new CleanupState(ex));
```

Comment:
this processing is already done on the level of `StreamingContext`, so you do not really change something :)